cli/internal/ffi/lockfile.go

```diff
@@ -0,0 +1,68 @@
+package ffi
+
+// #include "bindings.h"
+//
+// #cgo LDFLAGS: -L${SRCDIR} -lturborepo_ffi
+// #cgo windows LDFLAGS: -lole32 -lbcrypt -lws2_32 -luserenv
+import "C"
+
+import (
+	"errors"
+
+	ffi_proto "github.com/vercel/turbo/cli/internal/ffi/proto"
+)
+
+// NpmTransitiveDeps returns the transitive external deps of a given package based on the deps and specifiers given
+func NpmTransitiveDeps(content []byte, pkgDir string, unresolvedDeps map[string]string) ([]*ffi_proto.LockfilePackage, error) {
+	return transitiveDeps(npmTransitiveDeps, content, pkgDir, unresolvedDeps)
+}
+
+func npmTransitiveDeps(buf C.Buffer) C.Buffer {
```

Comment:
Go type system? For whatever reason `C.npm_transitive_closure` isn't of type `func(C.Buffer) C.Buffer` so I just manually wrap it and Go compiles

Reply:
Very odd. Would cleaning the build cache maybe fix this? Logically it seems fine but I'm paranoid of leaving it in because it is so innocuous looking.

Reply:
Clearing build cache doesn't help. Looking at the error message it looks like cgo functions are a different type than regular Go functions which isn't too surprising:
`cannot assign C.npm_transitive_closure (cgo function of type func(p0 _Ctype_struct_Buffer) (r1 _Ctype_struct_Buffer)) to func(_Ctype_struct_Buffer) _Ctype_struct_Buffer`

I've looked at it doesn't appear like there's a good way to describe taking a cgo function as a parameter in Go.

Reply:
Haha gotta love Go